### ISSUE TITLE
About detail

### ISSUE THREAD

**Issue by yzz9048** [SJTU-DMTai/OnlineTSF#5]
Interesting work!
I would appreciate that if you can tell me why there is"+1" in equation(3)

![Image](https://github.com/user-attachments/assets/58276031-4fdc-4dca-ab9b-daae59d644df)

And where are this implement and parameter update(eq. 4) exactly in your codes?

Hope to receive your reply，best wishes

**Comment by MogicianXD** [comment]
Thanks for your interest!

We note that $W_2$ is initialized as zeros, and the coefficients $\alpha$ and $\beta$ are initialized as ones. As a result, we initially make no adjustment to the parameter weights (see our Equation (4)). This ensures that, in the early stage when training our adapter, the model parameters begin from the pre-trained weights, giving normal predictions. Otherwise, if we randomly initialize the coefficients, the pretrained weights are perturbed to random values, and thereby the adapter takes more training time for convergence.

The code implementation can be found in [down_up.py Line 59-60](https://github.com/SJTU-DMTai/OnlineTSF/blob/466b682e08ce1bd676f783d7fa74efae21651258/adapter/module/down_up.py#L59)

By the way, we mistakenly put hats over $alpha$ and $beta$ in Equation (3). We will remove the hats and rectify notations in our arxiv version. Thanks for your comment!

**Comment by yzz9048** [comment]
> Thanks for your interest!
> 
> We note that W 2 is initialized as zeros, and the coefficients α and β are initialized as ones. As a result, we initially make no adjustment to the parameter weights (see our Equation (4)). This ensures that, in the early stage when training our adapter, the model parameters begin from the pre-trained weights, giving normal predictions. Otherwise, if we randomly initialize the coefficients, the pretrained weights are perturbed to random values, and thereby the adapter takes more training time for convergence.
> 
> The code implementation can be found in [down_up.py Line 59-60](https://github.com/SJTU-DMTai/OnlineTSF/blob/466b682e08ce1bd676f783d7fa74efae21651258/adapter/module/down_up.py#L59)
> 
> By the way, we mistakenly put hats over a l p h a and b e t a in Equation (3). We will remove the hats and rectify notations in our arxiv version. Thanks for your comment!


Thanks for your reply，I think this realization is reasonable.
One more question, I'm not clear why an error occurs on Line 104 of exe_basic.py when using Exp_Proceed in run.py. The error message is as follows。

![Image](https://github.com/user-attachments/assets/979bdfdb-8504-49c1-ba31-f838c9ef8655)

In-depth debugging revealed that when step==85 in adam.py, lr became None. I don't understand why this is and how could I fix this.

Additionally, I noticed that your code supports multi-card training (as can be seen there is '--use_multi_gpu'), however, the command on line 200 of run.py, args.gpu = args.device_ids[0], doesn't seem to use multiple cards?

**Comment by MogicianXD** [comment]
Can you print your all arguments from the argparser?

**Comment by yzz9048** [comment]
> Can you print your all arguments from the argparser?
here it is
![Image](https://github.com/user-attachments/assets/87ad0ebe-cf45-4b61-a0cb-56a5645b6f0b)

**Comment by MogicianXD** [comment]
You did not set a value for `args.online_learning_rate`, of which the default value is None. It would be desirable to set different learning rates for the offline and online phases

**Comment by yzz9048** [comment]
> You did not set a value for `args.online_learning_rate`, of which the default value is None. It would be desirable to set different learning rates for the offline and online phases

It does work! I set args.online_learning_rate to 0.001, I'm not sure if this is appropriate. And what about making multi-card training works?

**Comment by MogicianXD** [comment]
> Additionally, I noticed that your code supports multi-card training (as can be seen there is '--use_multi_gpu'), however, the command on line 200 of run.py, args.gpu = args.device_ids[0], doesn't seem to use multiple cards?

`CUDA_VISIBLE_DEVICES` must have multiple ids (see [here](https://github.com/SJTU-DMTai/OnlineTSF/blob/466b682e08ce1bd676f783d7fa74efae21651258/exp/exp_basic.py#L30)), while we only need one device to put our model in each subprocess. The inter-gpu transfer will be handled by `torch.distributed`.

Some script could be: `CUDA_VISIBLE_DEVICES=0,1,2,3 torchrun --nproc_per_node=4 run.py --use_multi_gpu`. No need to use `args.devices`.

By the way, we support the multi-gpu setting for model pretraining and the naive online learning method. We have not implemented a multi-gpu version of our Proceed method.

**Comment by MogicianXD** [comment]
> I set args.online_learning_rate to 0.001, I'm not sure if this is appropriate. 

You can refer to our scripts. The online learning rate depends on how long the online period is and how non-stationary the time series is. It would be desirable to frequently adjust the online learning according to the online performance. In the future, we plan to implement a method to automatically and dynamically decide the online learning rate.

**Comment by yzz9048** [comment]
> You can refer to our scripts. The online learning rate depends on how long the online period is and how non-stationary the time series is. It would be desirable to frequently adjust the online learning according to the online performance. In the future, we plan to implement a method to automatically and dynamically decide the online learning rate.

excellent! Thanks for detailed reply
